editor.planx.uk/src/pages/FlowEditor/lib/store/settings.ts

```diff
@@ -4,25 +4,24 @@ import { client } from "lib/graphql";
 import { FlowSettings, GlobalSettings, TextContent } from "types";
 import type { StateCreator } from "zustand";
 
+import { SharedStore } from "./shared";
+import { TeamStore } from "./team";
+
 export interface SettingsStore {
   flowSettings?: FlowSettings;
   setFlowSettings: (flowSettings?: FlowSettings) => void;
   globalSettings?: GlobalSettings;
   setGlobalSettings: (globalSettings: GlobalSettings) => void;
-  updateFlowSettings: (
-    teamSlug: string,
-    flowSlug: string,
-    newSettings: FlowSettings
-  ) => Promise<number>;
+  updateFlowSettings: (newSettings: FlowSettings) => Promise<number>;
   updateGlobalSettings: (newSettings: { [key: string]: TextContent }) => void;
 }
 
 export const settingsStore: StateCreator<
-  SettingsStore,
+  SettingsStore & TeamStore & SharedStore,
```

Comment:
I'm not quite understanding where the dependency for the `SharedStore` is coming from. Could you explain what this does?
Is it perhaps to get the `flowSlug` from `get()` below? 
Does including the shared store have any meaningful overhead?

Reply:
That's correct - it's so that we can just grab `flowSlug` as opposed to passing it into the function.

My understanding is as follows - 
 - All the stores are merged in `editor.planx.uk/src/pages/FlowEditor/lib/store/index.ts`
 - This change just allows TypeScript to know it has access to these variables which are controlled in another store - it's the type of the "getter" function essentially